cdc/owner/feed_state_manager.go

```diff
@@ -0,0 +1,281 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package owner
+
+import (
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/ticdc/cdc/model"
+	"github.com/pingcap/ticdc/pkg/filter"
+	"go.uber.org/zap"
+)
+
+type feedStateManager struct {
+	state         *model.ChangefeedReactorState
+	shouldRunning bool
+
+	adminJobQueue []*model.AdminJob
+}
+
+func (m *feedStateManager) Tick(state *model.ChangefeedReactorState) {
+	m.state = state
+	m.shouldRunning = true
+	defer func() {
+		if m.shouldRunning {
+			m.patchState(model.StateNormal)
+		} else {
+			m.cleanUpInfos()
+		}
+	}()
+	if m.state.Status == nil {
+		return
+	}
+	if pendingJobs := m.handleAdminJob(); pendingJobs {
+		return
+	}
+	switch m.state.Info.State {
+	case model.StateStopped, model.StateFailed, model.StateRemoved, model.StateFinished:
+		m.shouldRunning = false
+		return
+	}
+	errs := m.errorReportByProcessor()
+	m.HandleError(errs...)
+}
+
+func (m *feedStateManager) ShouldRunning() bool {
+	return m.shouldRunning
+}
+
+func (m *feedStateManager) MarkFinished() {
+	if m.state == nil {
+		// when state is nil, it means that Tick is never not called
+		// skip this and wait next tick to finish the changefeed
+		return
+	}
+	m.pushAdminJob(&model.AdminJob{
+		CfID: m.state.ID,
+		Type: model.AdminFinish,
+	})
+}
+
+func (m *feedStateManager) PushAdminJob(job *model.AdminJob) {
+	switch job.Type {
+	case model.AdminStop, model.AdminResume, model.AdminRemove:
+	default:
+		log.Panic("can not handle this job", zap.String("changefeedID", m.state.ID),
```

Comment:
```suggestion
		log.Panic("Can not handle this job", zap.String("changefeedID", m.state.ID),
```
It's better to capitalize the first letter of the log statement.